orttraining/orttraining/test/optimizer/graph_transform_test.cc

```diff
@@ -1394,6 +1394,99 @@ TEST_F(GraphTransformationTests, ScaledSumFusionThreeInputs) {
   }
 }
 
+/*
+Test graph as below.
+      graph input [1, 1, 256, 256] (float)  scalar_0   graph input [1, 1, 256, 256] (float)
+                                         \   /          |
+                                           Div         Div -- scalar_1
+[1, 1, 256, 256] (float)  scalar_3           \         /
+                \           /                  Add
+                      Sub                       /
+                        \                     /
+                          \                /
+                                Add
+                                 |
+                               Identity
+                                 |
+                graph out [1, 1, 256, 256] (float)
+
+*/
+TEST_F(GraphTransformationTests, ScaledSumFusionThreeInputs_LastAddNotHaveScaleInput) {
+  auto pre_graph_checker = [](Graph& graph) -> Status {
+    auto op_count_pre = CountOpsInGraph(graph);
+    TEST_RETURN_IF_NOT(op_count_pre.size() == 4U);
+    TEST_RETURN_IF_NOT(op_count_pre["Div"] == 2);
+    TEST_RETURN_IF_NOT(op_count_pre["Add"] == 2);
+    TEST_RETURN_IF_NOT(op_count_pre["Identity"] == 1);
+    TEST_RETURN_IF_NOT(op_count_pre["Sub"] == 1);
+    TEST_RETURN_IF_NOT(graph.GetAllInitializedTensors().size() == 3U);
+    return Status::OK();
+  };
+
+  auto post_graph_checker = [](Graph& graph) {
+    auto op_count = CountOpsInGraph(graph);
+    TEST_RETURN_IF_NOT(op_count.size() == 3U);
+    TEST_RETURN_IF_NOT(op_count["com.microsoft.ScaledSum"] == 1);
+    TEST_RETURN_IF_NOT(op_count["Identity"] == 1);
+    TEST_RETURN_IF_NOT(op_count["Sub"] == 1);
+
+    for (auto& node : graph.Nodes()) {
+      if (node.OpType() == "com.microsoft.ScaledSum") {
+        TEST_RETURN_IF_NOT(node.InputDefs().size() == 3U);
+
+        auto& attrs = node.GetAttributes();
+        TEST_RETURN_IF_NOT(attrs.find("scale_0") != attrs.end());
+        TEST_RETURN_IF_NOT(attrs.find("scale_1") != attrs.end());
+        TEST_RETURN_IF_NOT(attrs.find("scale_2") != attrs.end());
+        TEST_RETURN_IF_NOT(0.5f == static_cast<int>(attrs.at("scale_0").f()));
+        TEST_RETURN_IF_NOT(0.3f == static_cast<int>(attrs.at("scale_1").f()));
+        TEST_RETURN_IF_NOT(0.2f == static_cast<int>(attrs.at("scale_2").f()));
```

Comment:
why scale_2 is not 1.0 ?

Reply:
Looks the test check branch is not executing. Fixed it.